Explorer/Assets/DCL/Navmap/NavmapSearchBarController.cs

```diff
@@ -0,0 +1,70 @@
+using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using DCL.PlacesAPIService;
+using DCL.UI;
+using System;
+using System.Threading;
+
+namespace DCL.Navmap
+{
+    public class NavmapSearchBarController : IDisposable
+    {
+        public event Action<string> OnResultClicked;
+
+        private readonly SearchBarView view;
+        private readonly IPlacesAPIService placesAPIService;
+        private readonly SearchResultPanelController searchResultPanelController;
+
+        private CancellationTokenSource cts;
+
+        public NavmapSearchBarController(SearchBarView view, SearchResultPanelView searchResultPanelView, IPlacesAPIService placesAPIService, IAssetsProvisioner assetsProvisioner)
+        {
+            this.view = view;
+            this.placesAPIService = placesAPIService;
+
+            searchResultPanelController = new SearchResultPanelController(searchResultPanelView, assetsProvisioner);
+            searchResultPanelController.OnResultClicked += ClickedResult;
+            view.inputField.onValueChanged.AddListener(OnValueChanged);
+            view.clearSearchButton.onClick.AddListener(ClearSearch);
+        }
+
+        private void ClickedResult(string coordinates)
+        {
+            OnResultClicked?.Invoke(coordinates);
+        }
+
+        private void OnValueChanged(string searchText)
+        {
+            if (string.IsNullOrEmpty(searchText) || searchText.Length < 3)
+                searchResultPanelController.Hide();
+
+            cts?.Cancel();

```

Comment:
Don't you want to make an extension for this? I've seen these 3 lines pretty frequently already